File: 791/b.cpp
        #include <iostream>
        #include "map"

        #define ll long long
        #define fastio ios::sync_with_stdio(false), cin.tie(0), cout.tie(0);
        using namespace std;

        int main() {
            fastio
            int n, q;
            cin >> n >> q;
            map<int, int> mp;
            ll  res = 0;
            for(int i = 1; i <= n; i++) {
                int x;
                cin >> x;
                res += x;
                mp[i] = x;
            }
            int last;
            while(q--) {
                int a;
                cin >> a;
                if(a == 1) {
                    int i, x;
                    cin >> i >> x;
                    if(mp.count(i)) {
                        res -= mp[i];
                    }else res -= last;
                    mp[i] = x;
                    res += x;
                    cout << res << "\n";
                }else {
                    int x;
                    cin >> x;
                    res = 1ll*x * n;
                    cout << res << "\n";
                    last = x;
                    mp.clear();
                }

            } 
        }

File: 791/b.py
#include <bits/stdc++.h>

using namespace std;

int main() {
        long long n, q;
        cin >> n >> q;
        vector<long long> v(n);
        for(int i = 0; i < n; i++) 
            cin >> v[i];
        while(q--) {
            int e;
            cin >> e;
            long long res = 0;
            if(e == 1) {
                long long x, y;
                cin >> x >> y;
                v[--x] = y;
                for(int i = 0; i < n; i++) 
                    res += v[i];
                cout << res << "\n";
            }
            if(e == 2) {  
                long long x; 
                cin >> x;
                for(int i = 0; i < n; i++) 
                    v[i] = x;
                for(int i = 0; i < n; i++) 
                    res += v[i];
                cout << res << "\n";
                
            }

    }

}